src/core/server/saved_objects/import/import_saved_objects.ts

```diff
@@ -95,6 +97,39 @@ export async function importSavedObjectsFromStream({
     }
   }
 
+  // If target workspace and assigned data source is exists, it will check whether
+  // the assigned data sources in the target workspace include the data sources of the imported saved objects.
+  if (workspaces && workspaces.length > 0 && assignedDataSources) {
+    const errors: SavedObjectsImportError[] = [];
+    const duplicateObjects: typeof collectSavedObjectsResult.collectedObjects = [];
+    for (const object of collectSavedObjectsResult.collectedObjects) {
+      try {
+        const referenceDSId = await findDataSourceForObject(object, savedObjectsClient);
```

Comment:
The collectSavedObjectsResult should have all the objects that construct the `import tree`, can we just do the validation against this object so we do not need to do `bulkGet` and construct the tree twice.